chainer/graph_optimizations/static_graph_utilities.py

```diff
@@ -0,0 +1,197 @@
+import inspect
+
+import chainer
+
+
+def static_code(*dec_args, **dec_kwargs):
+    """Decorator to mark a function for inclusion in the static schedule.
+
+    This decorator is used to mark a function or method to be included
+    in the static schedule. This will only occur if the function is
+    called (directly or deeply) from inside a static chain's `__call__()`
+    method. That is, if a chain's `__call__()` uses the `@static_graph`
+    decorator, then any code that is executed while inside `__call__()`
+    that uses `@static_code` will be included in the corresponding
+    static schedule. Such code will be added to the static schedule in the
+    order that it was called.
+
+    This decorator should be applied to any code that needs to run each
+    iteration. This should ideally only include the code that performs
+    the actual forward and/or backward computations, and not include code
+    for initializaing parameters, checking types, etc.. If the user would
```

Comment:
etc. ~~If the user would like to...~~ To include other code in a static chain's `__call__()` method
to run every iteration, it should also use this
decorator.